backend/src/main/java/gov/cdc/usds/simplereport/service/OktaService.java

```diff
@@ -0,0 +1,121 @@
+package gov.cdc.usds.simplereport.service;
+
+import org.springframework.stereotype.Service;
+
+import java.util.Map; 
+import java.util.HashMap;
+
+import org.springframework.beans.factory.annotation.Value;
+
+import com.okta.sdk.client.Client;
+import com.okta.sdk.client.Clients;
+import com.okta.sdk.resource.user.User;
+import com.okta.sdk.resource.user.UserBuilder;
+import com.okta.sdk.resource.group.GroupList;
+import com.okta.sdk.resource.group.GroupType;
+import com.okta.sdk.resource.group.GroupBuilder;
+import com.okta.sdk.authc.credentials.TokenClientCredentials;
+
+import gov.cdc.usds.simplereport.db.model.ApiUser;
+
+import gov.cdc.usds.simplereport.config.authorization.OrganizationRole;
+
+/**
+ * Handles all user/organization management in Okta
+ */
+@Service
+public class OktaService {
+
+    @Value("${okta.entity-mgmt.org-url}")
+    private String ORG_URL;
+    @Value("${OKTA_API_KEY}")
+    private String API_TOKEN;
+    @Value("${simple-report.authorization.role-prefix}")
+    private String ROLE_PREFIX;
+
+    Client _client;
+
+
+    public OktaService() {
+        _client = Clients.builder()
+            .setOrgUrl(ORG_URL) 
+            .setClientCredentials(new TokenClientCredentials(API_TOKEN))
+            .build();
+    }
+
+    public void createUser(ApiUser apiUser) {
+        Map<String,Object> userProfileMap = new HashMap<String,Object>();
+        userProfileMap.put("firstName",apiUser.getNameInfo().getFirstName());
+        userProfileMap.put("middleName",apiUser.getNameInfo().getMiddleName());
+        userProfileMap.put("lastName",apiUser.getNameInfo().getLastName());
+        // This is cheating. Suffix and honorific suffix aren't the same thing. Shhh.
+        userProfileMap.put("honorificSuffix",apiUser.getNameInfo().getSuffix());
+        // We assume login == email
+        userProfileMap.put("email",apiUser.getLoginEmail());
+        userProfileMap.put("login",apiUser.getLoginEmail());
+
+        UserBuilder.instance()
+                .setProfileProperties(userProfileMap)
+                .buildAndCreate(_client);
+    }
+
+    public void updateUser(String oldUsername, ApiUser apiUser) {
+        User user = _client.listUsers(oldUsername, null, null, null, null).single();
+        user.getProfile().setFirstName(apiUser.getNameInfo().getFirstName());
+        user.getProfile().setMiddleName(apiUser.getNameInfo().getMiddleName());
+        user.getProfile().setLastName(apiUser.getNameInfo().getLastName());
+        // Is it our fault we don't accommodate honorific suffix? Or Okta's fault they 
+        // don't have regular suffix? You decide.
+        user.getProfile().setHonorificSuffix(apiUser.getNameInfo().getSuffix());
+        // We assume login == email
+        user.getProfile().setEmail(apiUser.getLoginEmail());
+        user.getProfile().setLogin(apiUser.getLoginEmail());
+        user.update();
+    }
+
+    public void createOrganization(String externalId, String name) {
```

Comment:
I would reverse the order of arguments here, to match the `Organization` constructor.